### PR TITLE
Respect takedowns on feedgen records, sync routes

### PR DESCRIPTION
**Pull request by devinivy**
With this change, takedowns issues on feedgen records will be applied to not surface high-level information about them through the API.  On sync routes, taken-down repos are now only served to admins and the taken-down account holder.

This one will be easiest to review with whitespace ignored 👍

### CODE REVIEW

File: packages/pds/src/api/com/atproto/sync/getBlocks.ts [hunk 1]
```diff
@@ -5,22 +5,34 @@ import SqlRepoStorage from '../../../../sql-repo-storage'
...
-      throw new InvalidRequestError(`Could not find cids: ${missingStr}`)
-    }
-    const car = blocksToCarStream(null, got.blocks)
+  server.com.atproto.sync.getBlocks({
+    auth: ctx.optionalAccessOrAdminVerifier,
+    handler: async ({ params, auth }) => {
+      const { did } = params
+      // takedown check for anyone other than an admin or the user
+      if (!isUserOrAdmin(auth, did)) {
+        const account = await ctx.services.account(ctx.db).getAccount(did, true)
```
Comment (dholms) [inline packages/pds/src/api/com/atproto/sync/getBlocks.ts:18]: thoughts on making a simpler method on this service that doesn't need any joins etc?

something like `getAccountState(did): TakenDown | NotExists | Available`

or `isTakenDown(did): boolean`

Comment (dholms) [reply packages/pds/src/api/com/atproto/sync/getBlocks.ts:18]: or actually `isRepoAvailable(did): boolean` may be best

File: packages/pds/src/api/com/atproto/sync/listRepos.ts [hunk 2]
```diff
@@ -2,15 +2,16 @@ import { InvalidRequestError } from '@atproto/xrpc-server'
...
 export default function (server: Server, ctx: AppContext) {
   server.com.atproto.sync.listRepos(async ({ params }) => {
     const { limit, cursor } = params
-    const ref = ctx.db.db.dynamic.ref
+    const { ref } = ctx.db.db.dynamic
     const innerBuilder = ctx.db.db
       .selectFrom('repo_root')
       .innerJoin('user_account', 'user_account.did', 'repo_root.did')
-      .where('repo_root.takedownId', 'is', null)
+      .where(notSoftDeletedClause(ref('repo_root')))
```
Comment (dholms) [inline packages/pds/src/api/com/atproto/sync/listRepos.ts:14]: lol this was def me, thx for fixing up 😛

**Review by dholms — APPROVED** [review]
just a note on a helper method, besides that looks great!

File: packages/pds/src/api/com/atproto/sync/getBlocks.ts (hunk 1 above)
Comment (devinivy) [reply packages/pds/src/api/com/atproto/sync/getBlocks.ts]: Just did that quick pass 👍

**Review by dholms — APPROVED** [review]

**Pull request merged**